### PR TITLE
fix(logged-in): use user service instead of cookie

### PR DESCRIPTION
**Pull request by JulieMarie**
### What's included?

1. Change authentication guard and interceptor to use the user-service to determine if logged in, instead of using the cookie.
2. Also added in work to handle redirect after log in. 

### Future work?

There is still documentation that needs to be done on vantage-ui-template for the redirect.

#### Test Steps

- [ ] `npm i`
- [ ] In `proxy.conf.js` file, change serverUrl to `https://appcenter.ux.ac.uda.io`
- [ ] `npm run serve`
- [ ] Go to `localhost:4200`
- [ ] Click `Log In`
- [ ] Instead of logging in, click the back arrow to go back to `localhost:4200`
- [ ] Verify that you don't show as logged in
- [ ] Click `Log In`
- [ ] Log in with `user1/user1`
- [ ] Go back to `localhost:4200` and verify that you show as logged in
- [ ] Click Log Out` and verify that you don't show as logged in

#### General Tests for Every PR

- [ ] `npm run serve:prod` still works.
- [ ] `npm run tslint` passes.
- [ ] `npm run stylelint` passes.
- [ ] `npm test` passes and code coverage is not lower.
- [ ] `npm run build:lib` still works.

closes https://jira.td.teradata.com/jira/browse/VUI-150

##### Screenshots or link to StackBlitz/Plunker
VUI-150

### CODE REVIEW

File: src/docs-ui/app/app.routes.ts [hunk 1]
```diff
@@ -4,11 +4,12 @@ import { MainComponent } from './main/main.component';
...
 
-const routes: Routes = [{
-    path: '', component: MainComponent, canActivate: [ VantageAuthenticationGuard ],
-  }, {
-    path: '**', redirectTo: '/',
-  },
+const routes: Routes = [
+  {
+    path: '', component: MainComponent, children: [
+      { path: 'login', component: MainComponent, canActivate: [ VantageAuthenticationGuard ] },
```
Comment (emoralesb05) [inline src/docs-ui/app/app.routes.ts]: Why did you add this as child route?

Comment (JulieMarie) [reply src/docs-ui/app/app.routes.ts]: For testing purposes to recreate the original issue that was on developer workspace.

File: src/lib/auth/guards/authentication.guard.ts [hunk 2]
```diff
@@ -1,30 +1,23 @@
...
-    let xsrfToken: string = this.getCookiebyName('XSRF-TOKEN');
-    if (!xsrfToken) {
-      window.location.href = '/start-login';
+    try {
+      // check the validity to see if already logged in
+      await this._sessionService.getInfo().pipe(timeout(5000)).toPromise();
+    } catch (e) {
+      // if not logged in, go ahead and log in...otherwise logout
+      // append the current path so we get redirected back upon login
+      (e.status === UNAUTHORIZED) ? window.location.href = `/start-login${this._router.url}` : this._sessionService.logout();
```
Comment (emoralesb05) [inline src/lib/auth/guards/authentication.guard.ts]: Does this work when deploying inside vantage?

Comment (JulieMarie) [reply src/lib/auth/guards/authentication.guard.ts]: Removed ${this._router.url} for now, but this needs to be fixed eventually.
